homestar-runtime/src/event_handler/swarm_event.rs

```diff
@@ -93,14 +97,178 @@ async fn handle_swarm_event<THandlerErr: fmt::Debug + Send, DB: Database>(
     event_handler: &mut EventHandler<DB>,
 ) {
     match event {
-        // TODO: add identify for adding compatable kademlia nodes.
-        // TODO: use kademlia to discover new gossip nodes.
+        SwarmEvent::Behaviour(ComposedEvent::Identify(identify_event)) => {
+            match identify_event {
+                identify::Event::Error { peer_id, error } => {
```

Comment:
@'ing @mriise

Reply:
Not thats easily matchable if any. Ideally we would do some verification with peers to ensure that we are adding a valid address, but thats not quite reliable. 

I also realized after thinking about it that this might be an issue of limits as well since we could add a whole bunch of addresses that might not be valid. 

If we want to be safe the easiest thing to do is not auto-add new external addresses, only trusting addresses we get from settings. I am more on that side of things at the moment if you are ok with that option as well.

Reply:
It's a good question. @mriise, when we add them as part of `identify:received` events, we should only be adding ones discovered using the protocol right, which should really only be homestar nodes. But, relatedly, we can't *always* trust a node even it advertises the same namespace. I think we can eventually have some way to validate things, as we won't be able to always configure nodes in settings. Maybe we can at least limit them in number for now?

Reply:
As discussed, we'll work on a limit (via settings) and add a comment to revisit more on validation/verification of these peers.